# NBA 2024-25: Utilizing Roles
## Notebook 04: Calculating All-Star Baselines (USG% and PRA)
This notebook flags players who were all-stars from the previous five seasons (2019-20 through 2023-24). It then calculates the **average Usage Rate (USG%)** and the **average PRA (Points + Rebounds + Assists)** for **every all-star during these five seasons**. These two averages are the "All-Star Baselines" that will become a major part of the later analysis.

In [2]:
# Import libraries
import pandas as pd
import numpy as np

___
## Load

In [4]:
# Load data
previous_5_seasons = pd.read_parquet(r"C:\Users\dylan\OneDrive\Documents\Portfolio_Projects\NBA_2024_25_utilizing_roles\03_python_outputs\AS_baselines\NBA_Per_Game_2019_2024.parquet")

In [5]:
# Calculate PRA (Points + Rebounds + Assists)
previous_5_seasons["PRA"] = previous_5_seasons["PTS"] + previous_5_seasons["TRB"] + previous_5_seasons["AST"]

In [6]:
# Confirm structure
previous_5_seasons.head(5)

Player  Age Team Pos   G  GS    MP    FG   FGA    FG%  \
0              Joel Embiid   29  PHI   C  39  39  33.6  11.5  21.8  0.529   
1              Luka Dončić   24  DAL  PG  70  70  37.5  11.5  23.6  0.487   
2    Giannis Antetokounmpo   29  MIL  PF  73  73  35.2  11.5  18.8  0.611   
3  Shai Gilgeous-Alexander   25  OKC  PG  75  75  34.0  10.6  19.8  0.535   
4            Jalen Brunson   27  NYK  PG  77  77  35.4  10.3  21.4  0.479   

   ...  BLK  TOV   PF   PTS                    Awards   PER    TS%  USG%  \
0  ...  1.7  3.8  2.9  34.7                        AS  34.1  0.644  39.6   
1  ...  0.5  4.0  2.1  33.9         MVP-3CPOY-6ASNBA1  28.1  0.617  36.0   
2  ...  1.1  3.4  2.9  30.4  MVP-4DPOY-9CPOY-12ASNBA1  29.9  0.649  33.0   
3  ...  0.9  2.2  2.5  30.1   MVP-2DPOY-7CPOY-3ASNBA1  29.3  0.636  32.8   
4  ...  0.2  2.4  1.9  28.7         MVP-5CPOY-5ASNBA2  23.4  0.592  32.5   

    Season   PRA  
0  2023-24  51.3  
1  2023-24  52.9  
2  2023-24  48.4  
3  2023-24  41.8  
4  2023-24  39.0  

[5 rows x 35 columns]

___
## Flag All-Stars

In [8]:
# Flag all-stars
previous_5_seasons["is_allstar"] = previous_5_seasons["Awards"].str.contains("AS", na=False)

In [9]:
# Value counts
previous_5_seasons["is_allstar"].value_counts()

is_allstar
False    2653
True      132
Name: count, dtype: int64

> There were **132 all-stars** from the previous 5 seasons.

___
## All-Star Baselines

In [12]:
# Filter for all-stars from previous 5 seasons
as_df = previous_5_seasons[previous_5_seasons["is_allstar"] == True].copy()

In [13]:
# Compute averages among these all-stars (and create a dictionary)
as_baseline = {
    "AS_USG_baseline": round(as_df["USG%"].mean(), 1),
    "AS_PRA_baseline": round(as_df["PRA"].mean(), 1)
}
as_baseline

{'AS_USG_baseline': 29.3, 'AS_PRA_baseline': 37.8}

> The average all-star USG% was 29.3% (per season).
> The average all-star PRA was 37.8 points + rebounds + assists (per game).

___
## Save

In [15]:
# Convert baseline dictionary to DataFrame
baseline_df = pd.DataFrame([
    {
        "Type": "All-Star",
        "USG_baseline": as_baseline["AS_USG_baseline"],
        "PRA_baseline": as_baseline["AS_PRA_baseline"]
    }
])
baseline_df

Type  USG_baseline  PRA_baseline
0  All-Star          29.3          37.8

In [16]:
# Save to parquet
baseline_df.to_parquet("NBA_Per_Game_2019_2024_baselines.parquet")